D3521R0.md

```diff
@@ -200,80 +200,46 @@ Given an example like:
 Conceptually: `try_cast<@*type*@>(@*subject*@)` is invoked, which returns an
 optional result that can be tested and dereferenced to match the *subpattern*.
 
-More precisely: Let the result of `try_cast<@*type*@>(@*subject*@)` be `__temp`
-where `try_cast` is looked up by ADL-only (same as `get` for structured bindings).
-The pattern matches if `__temp` contextually converts to `true` and
+More precisely: Let `E` be
+`std::try_cast<std::remove_cv_t<@*type*@>>(@*subject*@)`.  The match succeeds
+if and only if: `E` is well-formed; `E` is implicitly convertible to
+`@*type*@`; `E` contextually converts to `true`; and
```

Comment:
I think we're mixing things. Split up the static and dynamic criteria.

Reply:
Why?  This is just an explainer for the library folks to see what `try_cast` is used for.